PiwikTracker/Tracker.swift

```diff
@@ -186,3 +189,30 @@ extension Tracker {
         queue(event: event(withCategory: category, action: action, name: name, value: value))
     }
 }
+
+extension Tracker {
+    /// Set a permanent custom dimension.
+    ///
+    /// Use this method to set a dimension that will be send with every event. This is best for Custom Dimensions in scope "Visit". A typical example could be any device information or the version of the app the visitor is using.
+    ///
+    /// For more information on custom dimensions visit https://piwik.org/docs/custom-dimensions/
+    ///
+    /// - Parameter value: The value you want to set for this dimension.
+    /// - Parameter index: The index of the dimension. A dimension with this index must be setup in the piwik backend.
+    public func set(value: String, forDimension index: Int) {
+        let dimension = CustomDimension(index: index, value: value)
+        remove(dimensionAt: dimension.index)
+        dimensions.append(dimension)
+    }
+    
+    /// Removes a previously set custom dimension.
+    ///
+    /// Use this method to remove a dimension that was set using the `set(value: String, forDimension index: Int)` method.
+    ///
+    /// - Parameter index: The index of the dimension.
+    public func remove(dimensionAt index: Int) {
```

Comment:
I see your point. I did it this way to not let the user have to create a CustomDimension object, or even bother about something like Custom dimensions. I didn't want to expose to many types. 
Do you have another idea how to name it?